code/modules/religion/burdened/psyker.dm

```diff
@@ -1,25 +1,39 @@
 /obj/item/organ/internal/brain/psyker
 	name = "psyker brain"
-	desc = "This brain is blue, split into two hemispheres, and has immense psychic powers. Why does that even exist?"
+	desc = "This brain is blue, split into two hemispheres, and has immense psychic powers. What kind of monstrocity would use that?"
 	icon_state = "brain-psyker"
 	actions_types = list(
 		/datum/action/cooldown/spell/pointed/psychic_projection,
-		/datum/action/cooldown/spell/charged/projectile_booster,
+		/datum/action/cooldown/spell/charged/psychic_booster,
 		/datum/action/cooldown/spell/forcewall/psychic_wall,
 	)
+	organ_traits = list(TRAIT_ANTIMAGIC_NO_SELFBLOCK)
 
 /obj/item/organ/internal/brain/psyker/Insert(mob/living/carbon/inserted_into, special, drop_if_replaced, no_id_transfer)
 	if(!istype(inserted_into.get_bodypart(BODY_ZONE_HEAD), /obj/item/bodypart/head/psyker))
 		return FALSE
 	. = ..()
 	inserted_into.AddComponent(/datum/component/echolocation)
+	inserted_into.AddComponent(/datum/component/anti_magic, antimagic_flags = MAGIC_RESISTANCE_MIND)
+
+/obj/item/organ/internal/brain/psyker/Remove(mob/living/carbon/removed_from, special, no_id_transfer)
+	. = ..()
+	qdel(removed_from.GetComponent(/datum/component/echolocation))
+	qdel(removed_from.GetComponent(/datum/component/anti_magic))
 
 /obj/item/bodypart/head/psyker
 	limb_id = BODYPART_ID_PSYKER
 	is_dimorphic = FALSE
 	should_draw_greyscale = FALSE
 	bodypart_traits = list(TRAIT_DISFIGURED, TRAIT_BALD, TRAIT_SHAVED, TRAIT_BLIND)
```

Comment:
since the blindness is now tied to the echolocation, do we need the psyker head to have TRAIT_BLIND anymore?

Reply:
yes